### PR TITLE
[addons] adjustments and fixes to new manager layout

### PR DESCRIPTION
**Pull request by tamland**
This tries to address some of the feedback from #5862 with some small changes to the layout and other improvements.
- "Hides" the dependencies/running listing in a "Manage" sub-menu.
- Removes 'System add-on' showing these in My addons and Required dependencies instead.
- Categorizes 'My addons' (optional)

### CODE REVIEW

**Comment by stefansaraev** [comment]
:+1: much better. thanks.

it would still be nice to have some sort of feedback when user clicks on "check for updates", as before, busydialog or some notification "repo update started". whatever you think makes sense

**Comment by da-anda** [comment]
maybe you can adress the following as part of this PR: I noticed yesterday, that when you have two repositories containing an add-on with the same ID, that you can't install the add-on version from a specific repo because Kodi will always grab the "newer" one. Like when add-on A is in our official repo and some third party repo and I browse only our official repo and install A from there, then I still get add-on A from the third party repo.
I know that multiple repos having the same add-on ID is a bigger issue, but for the time beeing at least this case maybe could be fixed?

As for the restructuring:
I'm not sure I like "Manage..." because the entire Addon-Manager is there to manage the addons (also we should already have a "manage..." label that could probably be reused as it has a similar context). @jjd-uk - any suggestion what else to use? "Administrate" also doesn't fit. "Advanced management" maybe?

**Comment by tamland** [comment]
@da-anda The help text says "Manage your installed addons - Browse for and install addon", so I guess "Manage..." would be the manage part (?). I don't think "addon manager" or variant of "manage" is used anywhere else in the gui; just some people calling it that sometimes. 

As for the install problems: iirc it's been like that since always. It might be a feature..

**Comment by jjd-uk** [comment]
To avoid bike shedding here I've posted my thoughts at http://forum.kodi.tv/showthread.php?tid=191816&page=3

**Comment by NedScott** [comment]
Here's a public thread for non-team members as well: http://forum.kodi.tv/showthread.php?tid=222593

I've added my feedback there.

File: addons/resource.language.en_gb/resources/strings.po [hunk 1]
```diff
@@ -12942,7 +12946,13 @@ msgctxt "#24134"
 msgid "We were unable to load your configured language. Please check your language settings."
 msgstr ""
 
-#empty strings from id 24135 to 24992
+#empty strings from id 24135 to 24991
+
+#: xbmc/addons/CGUIViewStateAddonBrowser.cpp
+#: xbmc/filesystem/AddonsDirectory.cpp
+msgctxt "#24992"
```
Comment (da-anda) [inline addons/resource.language.en_gb/resources/strings.po]: I think you can reuse label 16106 here

File: addons/resource.language.en_gb/resources/strings.po [hunk 2]
```diff
@@ -12711,7 +12711,11 @@ msgctxt "#24087"
 msgid "All repositories"
 msgstr ""
 
-#empty strings id 24088
+#. Used in add-on browser
+#: xbmc/filesystem/AddonsDirectory.cpp
+msgctxt "#24088"
```
Comment (da-anda) [inline addons/resource.language.en_gb/resources/strings.po]: we might be able to reuse label 593 - at least the context should be similar enough

File: addons/resource.language.en_gb/resources/strings.po (hunk 1 above)
Comment (tamland) [inline addons/resource.language.en_gb/resources/strings.po]: 16106 is the context menu for 'managing' a video. This is about something different. You shouldn't reuse strings like that. The other one seems generic enough, so maybe..

Comment (MartijnKaijser) [inline addons/resource.language.en_gb/resources/strings.po]: the manage... for video is generic enough. Though i still dislike (against) using "manage..." at all.

Comment (tamland) [inline addons/resource.language.en_gb/resources/strings.po]: Management? Feel free to suggest some you don't dislike

Comment (da-anda) [inline addons/resource.language.en_gb/resources/strings.po]: reusing is OK as long as the context it its used for is the same. In doubt reuse it and make the context clearer by adding an according contextual note to the label.

**Comment by tamland** [comment]
All good then everyone?
jenkins build this please

**Comment by MartijnKaijser** [comment]
Ok for me if code is reviewed

**Comment by NedScott** [comment]
The only thing I have a strong opinion about is that we still don't have anything that says when the last add-on check was performed. I don't think cosmetics for the update button should justify a loss of functionality. Maybe a compromise would be to have a fake line say when an update was performed, like in v14, but not have it behave as an update button, so that the update button is still in the sidebar?

Everything else is a great improvement, thank you.

**Comment by phil65** [comment]
Definitely no fake line, we should try to stay consistent in terms of UI. 
Perhaps adding some new InfoLabel so skinners can include that info if wanted, but that would be a "nice-to-have" and does not have to be part of this PR.

**Comment by NedScott** [comment]
It's not a block for this specific PR, but "fixing" the regression is a must for v15, IMO. Even if it is an ugly fix in terms of UI, as it would still be on par with v14. When it comes to the functionality of a settings/management area (rather than the "normal" UI), and we are limited on time and not breaking APIs for v15, then we should choose function over style.

**Comment by Montellese** [comment]
IMO the "missing" time of the last update is neither a bug nor a must. TBH ever since that button was added to the main list I have been using it way too often simply because it's there. The information that the last update was 10 minutes ago didn't stop me from using the button again even though it made absolutely no sense. Therefore I think moving the button to the sidebar is a good choice and I doubt that showing the last update time is of much value. But such things should be discussed in the forum.

**Comment by NedScott** [comment]
Unfortunately, the existing forum threads don't seem to get any attention from those who are able to make these changes. I know how sacrilegious it is to make off-topic comments on github, but it the comments here always have a significantly higher impact than on the forums. I will try to make my case on the forum, but I will make one last plea on github:

I think my proposed compromise actually addresses the issue that Montellese brings up. The "fake line" wouldn't be a button anymore, but just an indicator. The button stays in the sidebar.

As for it's usefulness, it's vital to users such as myself who disable automatic update installs (on any system or program, I feel automatic updates is asking for a disaster). IIRC, the ability to see when the last check was done was a major sticking point for being able to only see notifications for new updates, rather than notifications + install.

There's always room for improvement, but the old system wasn't actually "broken", and now we're told to suck up a loss of a feature just because other people don't use it. It's not a "must", but why is this request being stonewalled? Is it so unreasonable to ask that changes made don't rip out something that other people do feel is very important, that they do actively use? This is like removing an option because someone doesn't like having an odd-number of choices.

**Comment by phil65** [comment]
Anyone dealing with UIs will break into tears if we start to put buttons only containing text next to clickable buttons. If something does not fit into the concept and is not utterly important (like this date label) then it should go. And btw, most package managers (all I tested, including "cluttered" ones like synaptic) do not show such info at an obvious place either.

**Comment by da-anda** [comment]
I agree, no button please. provide this info via infomanager and leave it up to the skins where to show. Also this info should be displayed next to the update check button, so having a list item for this wrong from a UX point anyways
@NedScott what I don't get in your scenario is in which way this info really matters. you for sure will remember if you checked for updates on this very day. and if you go to the addon manager because you intend to check for updates, does this info really prevent you from triggering the update check button if the last check was 1-5 days ago? Or what's the timespan you care to check for updates?

**Comment by HitcherUK** [comment]
How about using the toast when running a check to display the last check time/date?

**Comment by Montellese** [comment]
Just add it as an info label and let the skinners decide where they want to put it. On Confluence it could be put next to the update button in some way.

**Comment by NedScott** [comment]
@phil65 @da-anda @Montellese 

I have replied here: http://forum.kodi.tv/showthread.php?tid=222593&pid=1983539#pid1983539

**Comment by NedScott** [comment]
I notice with these changes the filters cannot be changed when I look at "my add-ons" and it seems weird that the filters can't be triggered until you enter a specific repo. I'm not seeing the benefit with that change.

**Comment by tamland** [comment]
Please continue the info label discussion in #6964.

@NedScott Several people have been confused by that, because it *is* very confusing that 'my addon' doesn't show all my addons and that you have to check side menu for filters. Secondly, 'Hide incompatible' lies. It has no effect outside of repositories (it never have) yet you can turn it on and off. It should have been grayed out.

**Comment by MartijnKaijser** [comment]
So can we get this in now the rest is of the discussion is split off? 
jenkins build this please

**Comment by NedScott** [comment]
@tamland you make an excellent point. It reminds me of how people get confused when "hide watched" for videos is accidentally turned on. Consider my previous statement withdrawn :)

**Comment by tamland** [comment]
atv2 failure unrelated

**Comment by MartijnKaijser** [comment]
@tamland i've got a feeling that some add-on categories are missing now from the addon manager.
one i found is "xbmc.python.library"

**Comment by tamland** [comment]
@MartijnKaijser That's correct. Dependencies are listed in System. They are hidden everywhere else.

**Pull request merged**